docs/decex/offline_operator_detection.md

```diff
@@ -0,0 +1,122 @@
+
+## Process
+
+1. Epoch start (immediately after previous epoch end)
+    - Apply staking rewards/updates.
+    - Determine eligible operators: operator_stake_i ≥ minimum_stake_units.
+    - Compute total_domain_stake and stake_share_i = operator_stake_i / total_domain_stake for each eligible operator.
+    - Record epoch_start_slot = current slot.
+    - Reset bundles_in_epoch_i = 0 for all eligible operators.
+    - Reset domain_block_count_in_epoch = 0.
+2. During the epoch
+    - On each bundle submission:
+        - For each included bundle (one per slot), increment bundles_in_epoch_i for its operator i by 1.
+3. End of epoch (evaluation) and immediate start of next epoch
+    - Compute slots_in_epoch S = epoch_end_slot − epoch_start_slot + 1.
+    - For each operator i that was eligible at this epoch’s start:
+        - Compute per-slot win probability: p_slot_i = min(1, (theta_num/theta_den) × stake_share_i).
+        - Compute expected bundles for the epoch: μ_i = S × p_slot_i.
+        - Throughput relevance enforcement:
+            - If μ_i < E_relevance: skip checking operator i (not throughput-relevant this epoch).
+            - Else:
+                - Chernoff lower-tail threshold:
+                    - r_i = floor( μ_i − sqrt( 2 × μ_i × ln(1/τ) ) ), clamped to [0, S].
+                    - Guarantee: P_honest[X_i < r_i] ≤ τ.
+                - Decision:
+                    - If x_i < r_i: mark operator i as underperforming, potentially offline
+                    - Else: operator i met expectations, producing at least the lower bound number of bundles in the
+                      epoch.
+    - Reset per-epoch counters:
+        - bundles_in_epoch_i = 0 for all operators.
+    - Start next epoch:
+        - Snapshot stakes, recompute eligibility and stake_share_i, set epoch_start_slot = current slot, and clear
+          exclusion flags by not including underperforming operators in the new eligible set.
+
+## Chernoff Threshold Details
+
+We test X_i ~ Binomial(S, p_slot_i) at lower tail level τ without scanning a CDF.
+
+- Mean: μ_i = S × p_slot_i
+- Additive Chernoff (lower tail): for any t ∈ [0, μ_i],
+  P[X_i ≤ μ_i − t] ≤ exp(−t^2 / (2 μ_i)).
+- Choose t = sqrt(2 μ_i ln(1/τ)), yielding:
+    - r_i = floor( μ_i − sqrt(2 μ_i ln(1/τ)) ), clamped to [0, S].
+    - This ensures an honest operator fails with probability at most τ.
+- Throughput relevance floor:
+    - When μ_i < 2 ln(1/τ), the bound becomes trivial (r_i ≤ 0). To avoid meaningless checks, enforce:
+        - E_relevance = ceil(max(E_base, 2 ln(1/τ))).
+        - Only operators with μ_i ≥ E_relevance are checked in the epoch.
+        - If k operators are “throughput-relevant” and each uses a Chernoff threshold calibrated at τ, then the
+          probability that all k honest operators fail in the same epoch is at most τ^k. For 4 operators, our current
+          operator set on mainnet, and τ = 1%,
+          that’s ≤ 10^-8 per epoch.
```

Comment:
This is good, but I just realised there's a huge catch:
```suggestion
          assuming all operators are running correctly, that’s ≤ 10^-8 per epoch.
```

If, for example, two operators were down, then the false positive probability is 10^-4 per epoch instead.